test/e2e/azure_test.go

```diff
@@ -844,6 +844,135 @@ var _ = Describe("Workload cluster creation", func() {
+						WaitForControlPlane:        e2eConfig.GetIntervals(specName, "wait-machine-upgrade"),
+						WaitForMachinePools:        e2eConfig.GetIntervals(specName, "wait-machine-pool-upgrade"),
+					}
+				})
+			})
+
+			By("Exercising machine pools", func() {
+				AKSMachinePoolSpec(ctx, func() AKSMachinePoolSpecInput {
+					return AKSMachinePoolSpecInput{
+						Cluster:       result.Cluster,
+						MachinePools:  result.MachinePools,
+						WaitIntervals: e2eConfig.GetIntervals(specName, "wait-machine-pool-nodes"),
+					}
+				})
+			})
+
+			By("creating a machine pool with public IP addresses from a prefix", func() {
+				// This test is also currently serving as the canonical
+				// "create/delete node pool" test. Eventually, that should be
+				// made more distinct from this public IP prefix test.
+				AKSPublicIPPrefixSpec(ctx, func() AKSPublicIPPrefixSpecInput {
+					return AKSPublicIPPrefixSpecInput{
+						Cluster:           result.Cluster,
+						KubernetesVersion: kubernetesVersion,
+						WaitIntervals:     e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
+					}
+				})
+			})
+
+			By("creating a machine pool with spot max price and scale down mode", func() {
+				AKSSpotSpec(ctx, func() AKSSpotSpecInput {
+					return AKSSpotSpecInput{
+						Cluster:           result.Cluster,
+						KubernetesVersion: kubernetesVersion,
+						WaitIntervals:     e2eConfig.GetIntervals(specName, "wait-worker-nodes"),
+					}
+				})
+			})
+
+			By("modifying nodepool autoscaling configuration", func() {
+				AKSAutoscaleSpec(ctx, func() AKSAutoscaleSpecInput {
+					return AKSAutoscaleSpecInput{
+						Cluster:        result.Cluster,
+						MachinePool:    result.MachinePools[0],
+						WaitIntervals:  e2eConfig.GetIntervals(specName, "wait-machine-pool-nodes"),
+						isClusterClass: true,
+					}
+				})
+			})
+
+			By("modifying additionalTags configuration", func() {
+				AKSAdditionalTagsSpec(ctx, func() AKSAdditionalTagsSpecInput {
+					return AKSAdditionalTagsSpecInput{
+						Cluster:       result.Cluster,
+						MachinePools:  result.MachinePools,
+						WaitForUpdate: e2eConfig.GetIntervals(specName, "wait-machine-pool-nodes"),
+					}
+				})
+			})
+
+			By("modifying the azure cluster-autoscaler settings", func() {
+				AKSAzureClusterAutoscalerSettingsSpec(ctx, func() AKSAzureClusterAutoscalerSettingsSpecInput {
+					return AKSAzureClusterAutoscalerSettingsSpecInput{
+						Cluster:       result.Cluster,
+						WaitIntervals: e2eConfig.GetIntervals(specName, "wait-control-plane"),
+					}
+				})
+			})
+
+			By("modifying node labels configuration", func() {
+				AKSNodeLabelsSpec(ctx, func() AKSNodeLabelsSpecInput {
+					return AKSNodeLabelsSpecInput{
+						Cluster:       result.Cluster,
+						MachinePools:  result.MachinePools,
+						WaitForUpdate: e2eConfig.GetIntervals(specName, "wait-machine-pool-nodes"),
+					}
+				})
+			})
+
+			By("modifying taints configuration", func() {
```

Comment:
Is it possible to share a good portion of this code with the existing AKS test?

Reply:
Added a common function in helpers.go: https://github.com/willie-yao/cluster-api-provider-azure/blob/clusterclass-aks-final/test/e2e/helpers.go#L946